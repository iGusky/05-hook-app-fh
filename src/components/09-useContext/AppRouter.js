import React from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>

                <NavBar />

               <div className="container">
                <Switch>
                        <Route path="/about" component={ AboutScreen } />
                        <Route path="/login" component={ LoginScreen } />
                        <Route exact path="/" component={ HomeScreen } />
                        {/* Se puede declarar un 404Component para una ruta qe no se encuentre */}
                        <Route component={ HomeScreen }/>
                </Switch>
               </div>

            </div>
        </Router>
    )
}
