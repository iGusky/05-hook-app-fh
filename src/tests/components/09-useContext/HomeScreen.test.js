import { mount, shallow } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen"
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Pruebas en <HomeScreen />', () => {

    const user = {
        name:'Gustavo',
        email: 'gus@mail.com'
    }

    const wrapper = mount ( 
        <UserContext.Provider value = {{
            user: user
        }} >
            <HomeScreen  />
        </UserContext.Provider>
    )

    test('debe mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    
})
