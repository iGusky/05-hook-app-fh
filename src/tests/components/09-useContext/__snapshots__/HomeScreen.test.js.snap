// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> debe mostrarse correctamente 1`] = `
Array [
  <h1>
    Home Screen
  </h1>,
  <hr />,
  <pre>
    {
   "name": "Gustavo",
   "email": "gus@mail.com"
}
  </pre>,
]
`;
